src/index.ts

```diff
@@ -433,12 +466,12 @@ export class Helper {
    * @returns {undefined} void
    * @memberof CouchbaseHelper
    */
-  public async set(topic: string, index: archivist.IArchivistIndex, data: any, options?: couchbase.UpsertOptions) {
+  public async set<ICouchbaseStateObject>(topic: string, index: archivist.IArchivistIndex, data: any, options?: couchbase.UpsertOptions) {
```

Comment:
Well, looks like I was wrong about the promise-in-a-promise (looks like `async` intelligently deals with that).

1. You can drop the `async` keyword from your code, the outcome is identical.
2. Alternatively, you can remove the Promise-part and just `return 'ok';` and the outcome is identical.